components/button/floating-buttons.js

```diff
@@ -0,0 +1,260 @@
+				position: relative;
+			}
+			.d2l-floating-buttons-container > div {
+				padding: 0.75rem 0 0 0;
+				position: relative;
+			}
+			.d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			.d2l-floating-buttons-inner-container ::slotted(button),
+			.d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-right: 0.75rem !important;
+				margin-bottom: 0.75rem !important;
+			}
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: 0.75rem !important;
+				margin-right: 0 !important;
+			}
+
+			@keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					transform: translate(0,0);
+				}
+			}
+			@-webkit-keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					-webkit-transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					-webkit-transform: translate(0,0);
+				}
+			}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this._container = null;
+		this._isRTL = false;
+		this._spacer = null;
+		this.minHeight = '500px';
+		this._containerClasses = {
+			'd2l-floating-buttons-container': true,
+			'd2l-floating-buttons-floating': false
+		};
+		this._innerContainerStyle = {
+			left: '',
+			right: '',
+			width: ''
+		};
+		this._buttonSpacerStyle = {
+			height: '',
+			display: ''
+		};
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		window.addEventListener('resize', () => {
+			this._reposition();
+		});
+		window.addEventListener('scroll', () => {
+			this._reposition();
+		});
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', () => {
+			this._reposition();
+		});
+		window.removeEventListener('scroll', () => {
+			this._reposition();
+		});
+	}
+
+	render() {
+		return html`
+			<div class=${classMap(this._containerClasses)} >
+				<div class="d2l-floating-buttons-inner-container"
+					style=${styleMap(this._innerContainerStyle)}>
+					<slot></slot>
+				</div>
+			</div>
+			<div class="d2l-floating-buttons-spacer"
+				style=${styleMap(this._buttonSpacerStyle)}>
+			</div>
+		`;
+	}
+
+	firstUpdated() {
+		this._reposition = this._reposition.bind(this);
+		this._container = this.shadowRoot.querySelector('.d2l-floating-buttons-container');
+		this._spacer = this.shadowRoot.querySelector('.d2l-floating-buttons-spacer');
+		this._isRTL = (getComputedStyle(this._container).direction === 'rtl');
```

Comment:
This can probably be improved by relying on the `RtlMixin` since we are now including it. The `dir` will be sprouted on the component host based on the document's `dir` unless a value is already specified by the consumer.  So this should be possible: `this.getAttribute('dir') === 'rtl'`.  Could also try using `this.dir === 'rtl'`. In such case, we can probably avoid the need for the `this._isRTL` member - it's purpose was to avoid costly `getComputedStyle` more than once.